pvlib/irradiance.py

```diff
@@ -1444,6 +1445,170 @@ def perez_driesse(surface_tilt, surface_azimuth, dhi, dni, dni_extra,
+                        )
+    except ValueError:
+        # this occurs when poa_error has the same sign at both end points
+        ghi = np.nan
+        conv = False
+        niter = -1
+    else:
+        ghi = result[0]
+        conv = result[1].converged
+        niter = result[1].iterations
+
+    return ghi, conv, niter
+
+
+def rtranspose_driesse_2023(surface_tilt, surface_azimuth,
+                            solar_zenith, solar_azimuth,
+                            poa_global,
+                            dni_extra=None, airmass=None, albedo=0.25,
+                            xtol=0.01,
+                            full_output=False):
+    '''
+    Estimate global horizontal irradiance (GHI) from global plane-of-array
+    (POA) irradiance.  This reverse transposition algorithm uses a bisection
+    search together with the continuous Perez-Driesse transposition and
+    continuous Erbs-Driesse decomposition models, as described in [1]_.
+
+    Parameters
+    ----------
+    surface_tilt : numeric
+        Panel tilt from horizontal. [degree]
+    surface_azimuth : numeric
+        Panel azimuth from north. [degree]
+    solar_zenith : numeric
+        Solar zenith angle. [degree]
+    solar_azimuth : numeric
+        Solar azimuth angle. [degree]
+    poa_global : numeric
+        Plane-of-array global irradiance, aka global tilted irradiance. [W/m^2]
+    dni_extra : None or numeric, default None
+        Extraterrestrial direct normal irradiance. [W/m^2]
+    airmass : None or numeric, default None
+        Relative airmass (not adjusted for pressure). [unitless]
+    albedo : numeric, default 0.25
+        Ground surface albedo. [unitless]
+    xtol : numeric, default 0.01
+        Convergence criterion.  The estimated GHI will be within xtol of the
+        true value. [W/m^2]
+    full_output : boolean, default False
+        If full_output is False, only ghi is returned, otherwise the return
+        value is (ghi, converged, niter). (see Returns section for details).
+
+    Returns
+    -------
+    ghi : numeric
+        Estimated GHI. [W/m^2]
+    converged : boolean, optional
+        Present if full_output=True. Indicates which elements converged
+        successfully.
+    niter : integer, optional
+        Present if full_output=True. Indicates how many bisection iterations
+        were done.
+
+    Notes
+    -----
+    Since :py:func:`scipy.optimize.bisect` is not vectorized, high-resolution
+    time series can be quite slow to process.
+
+    References
+    ----------
+    .. [1] A. Driesse, A. Jensen, R. Perez, A Continuous Form of the Perez
+        Diffuse Sky Model for Forward and Reverse Transposition, accepted
+        for publication in the Solar Energy Journal.
+
+    See also
+    --------
+    perez_driesse
+    erbs_driesse
+    gti_dirint
+    '''
+    # Contributed by Anton Driesse (@adriesse), PV Performance Labs. Nov., 2023
+
+    rtranspose_array = np.vectorize(_rtranspose)
```

Comment:
Could square it and find the minimum. Or does `scipy.optimize.root` handle this case? Grousing about scipy's documentation: I don't know what a "vector function" is - is it a scalar function of several variables, or is it a vector-valued function?

Reply:
`scipy.root` is vectorized, i.e., for vector-valued functions of a single variable. It may also be vectorized for multivariable functions, I didn't cook up an example.

```
import numpy as np
from scipy.optimize import root


def f(x, a):
    return (x + a - 1)**2


x0 = np.array([1, 1])
args = np.array([2, 3])
res = root(f, x0, args=args)
print(res.x)
```

Reply:
We can replace `bisect` with `root` in a follow-on PR.

Reply:
`scipy.root` would try to solve this as one large problem rather than a collection of smaller problems.

Reply:
I don't think that's correct. `root` finds a vector, each entry `res.x[i]` is the root of the function defined by `f` and `a[i]`. So it solves a vector of smaller problems. In what sense is it one large problem?

Reply:
It would be solved as a large problem because any element in the input vector could affect any or all values in the output vector.
>Methods hybr and lm in [root](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.root.html#scipy.optimize.root) cannot deal with a very large number of variables (N), as they need to calculate and invert a dense N x N Jacobian matrix on every Newton step. This becomes rather inefficient when N grows.

But there's no harm in trying it...

Reply:
I tried using `root` for `f_i(x) = 0, i=1, ..., 10000` and had a memory issue.

Reply:
Dropping the performance concern for now.